cli/src/version.rs

```diff
@@ -189,23 +190,23 @@ pub fn needs_updating() -> Result<CLIVersionDiff, Box<dyn Error + 'static>> {
     })
 }
 
-pub(crate) fn background_check_for_updates() -> mpsc::Receiver<Version> {
+pub(crate) fn background_check_for_updates(cdn_host: &str) -> mpsc::Receiver<Version> {
     debug!("Checking to see if there is a latest version");
     let (sender, receiver) = mpsc::channel();
 
     // don't check for updates if APOLLO_UPDATE_CHECK_DISABLED is set
-    if env::var("APOLLO_UPDATE_CHECK_DISABLED").is_ok() {
-        return receiver;
```

Comment:
oh so retunring this early was a source of the bugs as well? Interesting!

Reply:
Ah, no, this was a design style choice. Generally speaking I try to avoid `return` statements, especially in this case where the returned value is the same in both branches.

I adopted this approach when I started coding Scala (and fwiw was an explicit style guide at Twitter) where it was the first language I wrote that didn't require the `return` keyword explicitly. Rust has the same thing.

The backing idea here, is that returning early hurts readability as a function's code grows, and creates many "exit points" that one needs to keep in their head. Instead, branching as needed and having a single exit point is usually easier to parse. I realize this may be subjective and I'm not phrasing it well :)

Also, I will say that in cases where we want to exit very early and the rest of the function is long, it's actually better to use return, for example:
```rust
fn doSomethingOnVector(v: Vec<u32>) {
  if v.is_empty() return;

  // .. do something long and complex on v
}
```